### PR TITLE
MM-27135: Deactivated users show in channel members dropdown

### PR DESCRIPTION
**Pull request by agnivade**
We filter the user list by only activated users.

### Ticket link

https://mattermost.atlassian.net/browse/MM-27135

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -15038,7 +15038,7 @@
     },
     "into-stream": {
       "version": "3.1.0",
-      "resolved": "http://registry.npmjs.org/into-stream/-/into-stream-3.1.0.tgz",
+      "resolved": "https://registry.npmjs.org/into-stream/-/into-stream-3.1.0.tgz",
```
Comment (Willyfrog) [inline package-lock.json:15041]: can we remove the package-lock from the change? it doesn't seem related to it.

File: components/popover_list_members/popover_list_members.jsx [hunk 2]
```diff
@@ -100,7 +100,8 @@ export default class PopoverListMembers extends React.PureComponent {
     render() {
         const isDirectChannel = this.props.channel.type === Constants.DM_CHANNEL;
 
-        const items = this.props.sortedUsers.map((user) => (
+        const activeUsers = this.props.sortedUsers.filter((user) => user.delete_at === 0);
```
Comment (Willyfrog) [inline components/popover_list_members/popover_list_members.jsx:103]: minor nitpick (non blocking): since 0 is a value used to detect a especific state and it is outside of the actual expected values, I'd change that into a named constant that expreses such meaning (something like "NOT_DELETED" or "ACTIVE")

File: package-lock.json (hunk 1 above)
Comment (agnivade) [reply package-lock.json:15041]: Sure, to give some context to this, every time I make a webapp change, my npm version makes this change. I had been reverting the change since the past few months, so thought of including it this time. I will revert.

File: components/popover_list_members/popover_list_members.jsx (hunk 2 above)
Comment (agnivade) [reply components/popover_list_members/popover_list_members.jsx:103]: IMO, this is fine not to be refactored. Actually "NOT_DELETED = 0" makes it even more confusing IMO.

Nevertheless, I will make the change. But I note that `delete_at === 0` is being used all over the code. We should change all of them to be consistent.

**Comment by Willyfrog** [comment]
i definitely agree that we should change it throughout the code, thanks for taking one of the first steps towards that :D

**Comment by amyblais** [comment]
@calebroseland Kind reminder to review for v5.26,

File: components/popover_list_members/popover_list_members.jsx (hunk 2 above)
Comment (calebroseland) [reply components/popover_list_members/popover_list_members.jsx]: IMHO, I think a predicate function could help with clarity:

```ts
const NOT_DELETED_TIMESTAMP = 0;

export function isActive({delete_at}: User): boolean {
    return delete_at === NOT_DELETED_TIMESTAMP;
}

//...

import {isActive} from 'somewhere/utils/user';

const activeUsers = this.props.sortedUsers.filter(isActive);

```

That way any confusion behind the `0` constant contained in a clear `isActive` function.

Comment (agnivade) [reply components/popover_list_members/popover_list_members.jsx]: Done. IMO, if we are refactoring this to a separate function there wasn't a need to separately write "NEVER = 0".
